### PR TITLE
fix(#2007): astro preview port retry

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Closes https://github.com/withastro/astro/issues/2007
- Borrows the `port` retry behavior from `astro dev` for `astro preview`

## Testing

Manually tested. Lookin good! 😎 

## Docs

Bug fix only

### CODE REVIEW

File: packages/astro/src/core/preview/index.ts [hunk 1]
```diff
@@ -38,38 +38,55 @@ export default async function preview(config: AstroConfig, { logging }: PreviewO
...
+        httpServer?.on('error', onError);
+      };
+
+      const onError = (err: NodeJS.ErrnoException) => {
+        if (err.code && err.code === 'EADDRINUSE') {
+          if (!showedPortTakenMsg) {
+            info(logging, 'astro', msg.portInUse({ port }));
+            showedPortTakenMsg = true; // only print this once
+          }
+          port++;
```
Comment (drwpow) [inline packages/astro/src/core/preview/index.ts:68]: Update `this.port` in class?

**Review by drwpow — APPROVED** [review]
LGTM! Only just now noticed we don’t update `this.port` in preview (which is useful in tests, etc.)

File: packages/astro/src/core/preview/index.ts (hunk 1 above)
Comment (natemoo-re) [reply packages/astro/src/core/preview/index.ts:68]: `preview` isn't class-based, it's just modifying the `let port = 3000`

Comment (drwpow) [reply packages/astro/src/core/preview/index.ts:68]: Oh my mistake. I wasn’t looking at the whole file; you’re right

**Pull request merged**